ClearComputeNode.ps1

```diff
@@ -0,0 +1,156 @@
+    $Service = Get-Service $NodeMgrServiceName -ErrorAction SilentlyContinue
+    if ($Service -ne $null) {
+        Stop-Service $NodeMgrServiceName -ErrorAction SilentlyContinue
+        sc.exe delete $NodeMgrServiceName
+    }
+}
+
+function Stop-Agent {
+    Write-Host "Stopping Agent and removing service..."
+    $AgentServiceName = "ContrailAgent"
+    $Service = Get-Service $AgentServiceName -ErrorAction SilentlyContinue
+    if ($Service -ne $null) {
+        Stop-Service $AgentServiceName -ErrorAction SilentlyContinue
+        sc.exe delete $AgentServiceName
+    }
+}
+
+function Stop-DockerDriver {
+    Write-Host "Stopping Docker Driver and removing service..."
+    $DockerDriverServiceName = "contrail-docker-driver"
+    $Service = Get-Service $DockerDriverServiceName -ErrorAction SilentlyContinue
+    if ($Service -ne $null) {
+        Stop-Service $DockerDriverServiceName -ErrorAction SilentlyContinue
+        sc.exe delete $DockerDriverServiceName
+    }
+    # Docker Driver may run as a service. Or not.
+    Stop-ProcessIfExists -ProcessName "contrail-windows-docker-driver"
+}
+
+function Disable-VRouterExtension {
+    Param (
+        [Parameter(Mandatory = $true)] [String] $AdapterName,
+        [Parameter(Mandatory = $true)] [String] $ForwardingExtensionName,
+        [Parameter(Mandatory = $true)] [String] $VMSwitchName
+    )
+
+    Write-Host "Disabling Extension..."
+    Disable-VMSwitchExtension -VMSwitchName $VMSwitchName -Name $ForwardingExtensionName -ErrorAction SilentlyContinue | Out-Null
+    Get-ContainerNetwork | Where-Object NetworkAdapterName -eq $AdapterName | Remove-ContainerNetwork -ErrorAction SilentlyContinue -Force
+    Get-ContainerNetwork | Where-Object NetworkAdapterName -eq $AdapterName | Remove-ContainerNetwork -ErrorAction SilentlyContinue -Force
+}
+
+function Remove-AllContainers {
+    Write-Host "Removing all containers..."
+    $Containers = docker ps -aq
+    $MaxAttempts = 3
+    $TimesToGo = $MaxAttempts
+    if ($Containers.length -gt 0) {
+        while ( $Containers -and $TimesToGo -gt 0 ) {
+            if($Containers) {
+                $Command = "docker rm -f $Containers"
+                Invoke-Expression -Command $Command
+            }
+            $Containers = docker ps -aq
+            $TimesToGo = $TimesToGo - 1
+        }
+    }
+}
+
+function Remove-NotUsedDockerNetworks {
+    Write-Host "Removing not used Docker networks..."
+    docker network prune --force
+}
+
+function Uninstall-Components {
+    Write-Host "Uninstalling components..."
+    @(get-wmiobject Win32_product `
+        -Filter "name='Agent' OR name='vRouter' OR name='vRouter utilities' OR name='Contrail Docker Driver'") `
+        | ForEach-Object { msiexec.exe /x $_.IdentifyingNumber /q }
+}
+
+function Remove-AllDockerImages {
+    Write-Host "Removing all Docker images..."
+    $Images = docker images -aq
+    if ($Images.length -gt 0) {
+        docker rmi $Images
+    }
```

Comment:
All these helper functions don't seem to have error reporting. Or are we relying on ErrorAction?